threema-connector/processes/util/messageEncryption.p.json

```diff
@@ -0,0 +1,75 @@
+{
+  "format" : "10.0.0",
+  "id" : "18B22CD369A6089F",
+  "kind" : "CALLABLE_SUB",
+  "config" : {
+    "data" : "threema.connector.receiverData"
+  },
+  "elements" : [ {
+      "id" : "f0",
+      "type" : "CallSubStart",
+      "name" : "call(receiverData)",
+      "config" : {
+        "callSignature" : "call",
+        "input" : {
+          "params" : [
+            { "name" : "receiverData", "type" : "threema.connector.receiverData" }
+          ],
+          "map" : {
+            "out" : "param.receiverData"
+          }
+        },
+        "result" : {
+          "params" : [
+            { "name" : "receiverData", "type" : "threema.connector.receiverData" }
+          ],
+          "map" : {
+            "result.receiverData" : "in"
+          }
+        }
+      },
+      "visual" : {
+        "at" : { "x" : 96, "y" : 64 },
+        "icon" : "res:/webContent/icons/threema-icon_black.png?small"
+      },
+      "connect" : { "id" : "f4", "to" : "f2" }
+    }, {
+      "id" : "f1",
+      "type" : "CallSubEnd",
+      "visual" : {
+        "at" : { "x" : 480, "y" : 64 }
+      }
+    }, {
+      "id" : "f2",
+      "type" : "Script",
+      "name" : "Encrypt Message",
+      "config" : {
+        "output" : {
+          "code" : [
+            "import ch.threema.apitool.results.EncryptResult;",
+            "import ch.threema.apitool.CryptTool;",
+            "import javax.xml.bind.DatatypeConverter;",
+            "",
+            "String privateKey = ivy.var.connector_privateKey;",
+            "String publicKey = in.publicKey;",
+            "String msg = in.plainMessage;",
+            "",
+            "Array<byte> encPrivKey = DatatypeConverter.parseHexBinary(privateKey);",
+            "Array<byte> encPubKey = DatatypeConverter.parseHexBinary(publicKey);",
+            "",
+            "CryptTool tool = new CryptTool();",
+            "EncryptResult result = tool.encryptTextMessage(msg, encPrivKey, encPubKey);",
+            "",
+            "out = in;",
+            "out.encryptedMessage = DatatypeConverter.printHexBinary(result.getResult());",
+            "out.nonce = DatatypeConverter.printHexBinary(result.getNonce());"
```

Comment:
isolate it in a javaclass and ... static utility method ... call it here with a one-liner